### ISSUE TITLE
Image Copy '--target-subscription' Argument Ignored

### ISSUE THREAD

**Issue by hwwilliams** [Azure/azure-cli-extensions#5663]
### Related command

```bash
az image copy --source-resource-group <source_rg_name> \
        --source-object-name <source_image_name> \
        --source-type image \
        --target-subscription <target_subscription_id> \
        --target-location <target_location> \
        --target-resource-group <target_rg_name>  \
        --target-name <target_image_name> \
        --temporary-resource-group-name <temp_rg_name>
```

### Extension name (the extension in question)
image-copy-extension

### Description of issue (in as much detail as possible)

My organization is using the `image-copy-extension` and we noticed on recent workflows that the extension is now ignoring the `--target-subscription` argument and seems to use the authentication context subscription as the target subscription for all temporary resource and final image creation instead.

We noticed this change because previously the temporary resource group we used was stored within the target subscription and this meant that the temporary storage account, blob, and snapshot all got created within this temporary resource group but now the extension seems to be looking for the temporary resource group in the default subscription associated with the authentication context instead of the target subscription provided to the `--target-subscription` argument.

My Python experience is still limited but I have looked at the [recent commit](https://github.com/Azure/azure-cli-extensions/commit/3a6621cc0c96aab5ddd2388f4a2e6a81f4ed2f0a) and I see that the `target_subscription` variable is now being set without checking if the argument has been passed and I'm not sure if this was changed for a reason or if this is just a bug that can be fixed by only setting the `target_subscription` variable if the argument is null.

![image](https://user-images.githubusercontent.com/20447447/209253966-f9129522-3988-4c73-9abc-d6fc0e7ef991.png)

Link to the line in question.

https://github.com/Azure/azure-cli-extensions/blob/42102c3a81ad23407e7d929521ea212805b4688d/src/image-copy/azext_imagecopy/custom.py#L37

There is also a debug line below the variable declaration that we've seen continues to display the subscription ID for the authentication context and not the target subscription we're providing as an argument.

-----

**Comment by hwwilliams** [comment]
Is there a way to pin the version of the `image-copy` extension we're using with Azure CLI while we wait for a potential fix?

The extension was working fine a few versions ago.

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

**Comment by zhoxing-ms** [comment]
@hwwilliams Your analysis is correct. This problem was introduced by #https://github.com/Azure/azure-cli-extensions/pull/5633/files#r1057132954. As we are in the Christmas holiday season, we plan to fix this issue in early January

**Comment by hwwilliams** [comment]
> @hwwilliams Your analysis is correct. This problem was introduced by #https://github.com/Azure/azure-cli-extensions/pull/5633/files#r1057132954. As we are in the Christmas holiday season, we plan to fix this issue in early January. May I ask if this time is in line with your expectations?

That should be fine as most of our team is also out until then as well. I assume there is no way to pin the older extension version until that time just in case it takes longer than estimated to fix?

**Comment by zhoxing-ms** [comment]
@hwwilliams You can use the `--version` parameter to pin the specific version when installing the extension, such as `az extension add --name anextension --version 1.0.0`

**Comment by hwwilliams** [comment]
> @hwwilliams You can use the `--version` parameter to pin the specific version when installing the extension, such as `az extension add --name anextension --version 1.0.0`

Thank you, that version pinning worked for now.

**Comment by yanzhudd** [comment]
Hello @hwwilliams, this bug has been fixed, and you could upgrade the version (0.2.12) of image-copy to apply the change:

`az extension update --name image-copy-extension`

Alternatively, remove and add this extension to update the version:
`az extension remove --name image-copy-extension`
`az extension add --name image-copy-extension`
